Source/Projections/Builder/TypedClassProjectionMethod.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using Dolittle.SDK.Async;
+using Dolittle.SDK.Events;
+
+namespace Dolittle.SDK.Projections.Builder
+{
+    /// <summary>
+    /// An implementation of <see cref="IProjectionMethod{TProjection}" /> that invokes a method on a projection instance for an event of a specific type.
+    /// </summary>
+    /// <typeparam name="TProjection">The <see cref="Type" /> of the projection.</typeparam>
+    /// <typeparam name="TEvent">The <see cref="Type" /> of the event.</typeparam>
+    public class TypedClassProjectionMethod<TProjection, TEvent> : IProjectionMethod<TProjection>
+        where TProjection : class, new()
+        where TEvent : class
+    {
+        readonly TaskResultProjectionMethodSignature<TProjection, TEvent> _method;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TypedClassProjectionMethod{TProjection, TEvent}"/> class.
+        /// </summary>
+        /// <param name="method">The <see cref="TaskProjectionMethodSignature{TProjection, TEvent}"/> method to invoke.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector" />.</param>
+        public TypedClassProjectionMethod(TaskProjectionMethodSignature<TProjection, TEvent> method, KeySelector keySelector)
+            : this(
+                (TProjection instanceAndReadModel, TEvent @event, ProjectionContext context)
+                    => method(instanceAndReadModel, @event, context).ContinueWith(_ => ProjectionResultType.Replace, TaskScheduler.Default),
```

Comment:
Same here as above - wondering if this behaves the same way as an async function with await in it.